hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java

```diff
@@ -58,7 +58,7 @@ public void preHandle(@NonNull final PreHandleContext context) {
         requireNonNull(context);
         final var op = context.getTxn().contractCreateInstanceOrThrow();
         final var adminKey = asHederaKey(op.adminKeyOrElse(Key.DEFAULT));
-        if (adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
+        if (op.hasAdminKey() && adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
```

Comment:
shouldn't `op.hasAdminKey` be present before line 60 ?

Reply:
I was thinking it'd be awkward to check, but I found a way that's not so bad. Thanks :)